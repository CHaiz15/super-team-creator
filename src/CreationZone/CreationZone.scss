@import '../variables';

.creation-fav-btn {
  font-family: 'Bangers';
  background-color: white;
  outline: none;
  border-radius: 0  0 20px 20px;
  font-size: 2em; 
  padding: 1%;
  margin: 18% 0 0 0;
  cursor: pointer;
  box-shadow: 0 3px black;
  animation: slide-top 0.5s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
}

.creation-fav-btn:hover {
  @include mainBtnHover();
}

.creation-fav-btn:active {
  @include mainBtnActive();
}

.creation-header {
  height: 10vh;
}

.creation-title {
  @include mainTitles();
  text-align: center;
}

.build-btn {
  @include mainBtn();
  margin: 15%;
}

.build-btn:hover {
  @include mainBtnHover();
}

.build-btn:active {
  @include mainBtnActive();
}

.creation-zone {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  height: 100vh;
}

.team-name-label {
  margin: 10% 0 0 0;
  color: white;
  font-size: 2em;
}

.team-name-input {
  width: 30%;
  font-size: 2.5em;
  border-radius: 5px;
  outline: none;
  box-shadow: 0 3px black;
}

.input-container {
  display: flex;
  flex-direction: column;
  margin: 0 2% 0 2%;
  width: 220px;
  justify-content: center;
}

.member-inputs {
  display: flex;
  margin: 5% 0 0 0;
  font-size: 2em;
  width: 100%;
  justify-content: center;
}

.member-label {
  font-size: 1em;
  color: white;
  text-align: center;
}

.member-input {
  width: 15rem;
  height: 2rem;
  font-size: 1em;
  border-radius: 5px;
  outline: none;
  box-shadow: 0 3px black;
}
