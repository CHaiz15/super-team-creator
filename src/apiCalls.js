export const getMember = (member) => {
  let proxyUrl = 'https://cors-anywhere.herokuapp.com/'
  return fetch(proxyUrl + `https://superheroapi.com/api/10222961455404600/search/${member}`)
  .then(res => {
    if(!res.ok) {
      throw Error('Something is not right, try again later')
    }
    return res.json()})
}