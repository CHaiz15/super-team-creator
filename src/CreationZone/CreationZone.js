import React, { Component } from 'react'
import './CreationZone.scss'
import { connect } from 'react-redux'
import { createTeam } from '../Actions'
import { getMember } from '../apiCalls'
import { NavLink } from 'react-router-dom'

export class CreationZone extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      memberOne: '',
      memberTwo: '',
      memberThree: '',
    }
  }

  handleChange(e) {
    e.preventDefault();
    this.setState({[e.target.name]: e.target.value})
  }

  fetchMembers = async () => {
    let member1, member2, member3;
    let response1 = await getMember(this.state.memberOne)
      if(response1.response !== 'error') {
        member1 = response1.results[0];
      }
    let response2 = await getMember(this.state.memberTwo)
      if(response2.response !== 'error') {
        member2 = response2.results[0];
      }
    let response3 = await getMember(this.state.memberThree)
      if(response3.response !== 'error') {
        member3 = response3.results[0];
      }
    
    let team = {
      name: this.state.name,
      memberOne: member1,
      memberTwo: member2,
      memberThree: member3,
    }

    this.props.createTeam(team)
  }

  render() {
    return (
      <>
        <section className='creation-zone'>
          <h1 className='creation-title'>Creation Zone</h1>
          <label className='team-name-label'>Team Name:</label>
          <input 
            name='name'
            value={this.state.name}
            className='team-name-input'
            placeholder='Team Name'
            onChange={(e) => this.handleChange(e)}
            maxLength='40'
          />
          <div className='member-inputs'>
            <label className='member-label'>Team Member 1:
            <input 
              name='memberOne'
              value={this.state.memberOne}
              className='member-input'
              placeholder='Team Member 1'
              onChange={(e) => this.handleChange(e)}
              maxLength='25'
            />
            </label>
            <label className='member-label'>Team Member 2:
            <input 
              name='memberTwo'
              value={this.state.memberTwo}
              className='member-input'
              placeholder='Team Member 2'
              onChange={(e) => this.handleChange(e)}
              maxLength='25'
            />
            </label>
            <label className='member-label'>Team Member 3:
            <input 
              name='memberThree'
              value={this.state.memberThree}
              className='member-input'
              placeholder='Team Member 3'
              onChange={(e) => this.handleChange(e)}
              maxLength='25'
            />
            </label>
          </div>
          <NavLink 
            style={this.state.name && this.state.memberOne && this.state.memberTwo && this.state.memberThree ? {'pointer-events': 'visible'}:{'pointer-events': 'none'}}
            className='build-btn' 
            type='button' 
            to='/team' 
            onClick={() => this.fetchMembers()}
            >Build
          </NavLink>
        </section>
      </>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  createTeam: team => dispatch(createTeam(team)),
})

export default connect(null, mapDispatchToProps)(CreationZone)