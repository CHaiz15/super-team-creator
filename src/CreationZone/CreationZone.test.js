import React from 'react'
import { CreationZone } from './CreationZone'
import { shallow } from 'enzyme'
import { mapDispatchToProps } from './CreationZone'
import { createTeam } from '../actions'
import { getMember } from '../apiCalls'

describe('CreationZone', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<CreationZone/>);
    })  
    it('should be an instance of the CreationZone component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    describe('mapDispatchToProps', () => {
      it('calls dispatch with a createTeam action when fetchMembers is called', () => {
        const mockTest = {
          name:'Best Team Ever',
          memberOne: {
            name:'Hulk', 
            powerstats: {
              Intelligence: 81,
              strength: 40,
              Speed: 29,
              Durability: 55,
              Power: 63,
              Combat: 90,
            },
            image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
          },
          memberTwo: {
            name:'Thanos',
            powerstats: {
              Intelligence: 81,
              strength: 40,
              Speed: 29,
              Durability: 55,
              Power: 63,
              Combat: 90,
            },
            image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
          },
          memberThree: {
            name:'Batman',
            powerstats: {
              Intelligence: 81,
              strength: 40,
              Speed: 29,
              Durability: 55,
              Power: 63,
              Combat: 90,
            },
            image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
          },
        }
        const mockDispatch = jest.fn()
        const actionToDispatch = createTeam(mockTest)

        const mappedProps = mapDispatchToProps(mockDispatch)
        mappedProps.createTeam(mockTest)

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      })

      describe('CreationZone React tests', () => {
      it('should start with default state values', () => {
        expect(wrapper.state('name')).toEqual('')
        expect(wrapper.state('memberOne')).toEqual('')
        expect(wrapper.state('memberTwo')).toEqual('')
        expect(wrapper.state('memberThree')).toEqual('')
      })

      it('should call handleChange method', () => {
        const mockEvent = {
          preventDefault: jest.fn(), 
          target: {name: 'name', value: 'Best Team Ever'
        }}
        wrapper.instance().handleChange = jest.fn()
        wrapper.instance().handleChange(mockEvent)
        expect(wrapper.instance().handleChange).toHaveBeenCalled()
      })

      it('should update state.name when handleChange is called', () => {
        const mockEvent = {
          preventDefault: jest.fn(), 
          target: {name: 'name', value: 'Best Team Ever'
        }}
        const expected = 'Best Team Ever'
        wrapper.instance().handleChange(mockEvent)
        expect(wrapper.state('name')).toEqual(expected)
      })

      it('should update state.name when handleChange is called', () => {
        const mockEvent = {
          preventDefault: jest.fn(), 
          target: {name: 'name', value: 'Best Team Ever'
        }}
        const expected = 'Best Team Ever'
        wrapper.instance().handleChange(mockEvent)
        expect(wrapper.state('name')).toEqual(expected)
      })

      it('should update state.memberOne when handleChange is called', () => {
        const mockEvent = {
          preventDefault: jest.fn(), 
          target: {name: 'memberOne', value: 'Ben 10'
        }}
        const expected = 'Ben 10'
        wrapper.instance().handleChange(mockEvent)
        expect(wrapper.state('memberOne')).toEqual(expected)
      })

      it('should update state.memberTwo when handleChange is called', () => {
        const mockEvent = {
          preventDefault: jest.fn(), 
          target: {name: 'memberTwo', value: 'Thanos'
        }}
        const expected = 'Thanos'
        wrapper.instance().handleChange(mockEvent)
        expect(wrapper.state('memberTwo')).toEqual(expected)
      })

      it('should update state.memberThree when handleChange is called', () => {
        const mockEvent = {
          preventDefault: jest.fn(), 
          target: {name: 'memberThree', value: 'Hulk'
        }}
        const expected = 'Hulk'
        wrapper.instance().handleChange(mockEvent)
        expect(wrapper.state('memberThree')).toEqual(expected)
      })

      describe('CreationZone React tests', () => {
        let wrapper;
        beforeEach(() => {
          wrapper = shallow(<CreationZone/>);
        })  
      })
      
      it('should call fetchMembers when Build button is clicked', () => {
        wrapper.instance().fetchMembers = jest.fn()
        wrapper.find('NavLink').simulate('click')
        expect(wrapper.instance().fetchMembers).toHaveBeenCalled()
      })
      
      it('should return an array of ideas (HAPPY)', () => {
        let mockResponse = {response: "success",
        'results-for': "Hulk",
        results: [{
            id: 332,
            name: "Hulk",
            powerstats: {
                intelligence: "88",
                strength: "100",
                speed: "63",
                durability: "100",
                power: "98",
                combat: "85"
            },
            image: {
            url: "https://www.superherodb.com/pictures2/portraits/10/100/125.jpg"
          }
        }]
      }
        expect(getMember()).resolves.toEqual(mockResponse)
      });
    })
  })
})

