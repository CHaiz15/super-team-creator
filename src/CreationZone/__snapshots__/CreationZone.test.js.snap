// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreationZone should be an instance of the CreationZone component 1`] = `ShallowWrapper {}`;
